### PR TITLE
Automate Icon File Generation (not ready for merge)

### PR DESCRIPTION
**Pull request by micahgodbolt**
Seeing as we have 3 separate files that are just different forms of the same data, it seemed to make sense to generate all 3 files from a single set of data. 

This PR still needs to update Gulp to properly create the temp files before Sass is compiled.

### CODE REVIEW

File: src/icons/data/icons.json [hunk 1]
```diff
@@ -0,0 +1,708 @@
+{
```
Comment (mikewheaton) [inline src/icons/data/icons.json]: This file should now be in the 'dist' folder, right?

**Comment by micahgodbolt** [comment]
{"name":"{name}", "value":"{value}"},

File: gulp/modules/Plugins.js [hunk 2]
```diff
@@ -35,6 +35,8 @@ var Plugins = function() {
 	this.path = require('path');
 	this.pkg = require('../../package.json');
 	this.fs = require('fs');
+	this.yaml = require('js-yaml');
+
```
Comment (battletoilet) [inline gulp/modules/Plugins.js]: nit: extra line and if we are not using this anymore let's remove it.

File: package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
     "gulp-tslint": "^7.0.1",
     "gulp-typescript": "^3.1.1",
     "gulp-util": "^3.0.1",
+    "js-yaml": "^3.7.0",
```
Comment (battletoilet) [inline package.json]: Same here let's remove this if it's not being used.

File: gulp/modules/Plugins.js (hunk 2 above)
Comment (micahgodbolt) [inline gulp/modules/Plugins.js]: good catch

**Comment by mikewheaton** [comment]
Closing this for now as it's not being actively worked on.

**Pull request closed without merging**